crates/flipperzero/src/alloc.rs

```diff
@@ -0,0 +1,55 @@
+use core::alloc::{GlobalAlloc, Layout};
+use core::ffi::c_void;
+
+use flipperzero_sys::c_string;
+use flipperzero_sys::furi;
+
+extern "C" {
+    #[link_name = "free"]
+    pub fn free(p: *mut c_void);
+
+    #[link_name = "aligned_malloc"]
+    pub fn aligned_malloc(size: usize, align: usize) -> *mut *mut c_void;
+
+    #[link_name = "realloc"]
+    pub fn realloc(p: *mut c_void, size: usize) -> *mut c_void;
+}
+
+pub struct FuriAlloc;
+
+unsafe impl GlobalAlloc for FuriAlloc {
+    #[inline]
+    unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
+        aligned_malloc(
+            layout.size(),
+            layout.align().max(core::mem::size_of::<usize>()),
+        ) as *mut u8
+    }
+
+    #[inline]
+    unsafe fn dealloc(&self, ptr: *mut u8, _layout: Layout) {
+        free(ptr as *mut c_void);
```

Comment:
Because `aligned_malloc` returns a pointer to the aligned byte, not the actual heap allocation, we have to pair it with [`aligned_free`](https://github.com/flipperdevices/flipperzero-firmware/blob/0f9ea925d30f7a6183abe1002d680e3d1613a159/furi/core/memmgr.c#L109).